server/api/users/me-auth/ctftime/patch.js

```diff
@@ -0,0 +1,34 @@
+const { responses } = require('../../../../responses')
+const database = require('../../../../database')
+const auth = require('../../../../auth')
+
+module.exports = {
+  method: 'patch',
+  path: '/users/me/auth/ctftime',
+  requireAuth: true,
+  schema: {
+    body: {
+      type: 'object',
+      properties: {
+        ctftimeToken: {
+          type: 'string'
+        }
+      },
+      required: ['ctftimeToken']
+    }
+  },
+  handler: async ({ req, user }) => {
+    const ctftimeData = await auth.token.getData(auth.token.tokenKinds.ctftimeAuth, req.body.ctftimeToken)
+    if (ctftimeData === null) {
+      return responses.badToken
```

Comment:
We should differentiate between a bad auth token and some other token that is sent being bad, since currently on the client we interpret a `badToken` as meaning that the auth token is bad then force the user to login again.